internal/validation/mock_metric_validation.go

```diff
@@ -15,9 +15,20 @@
 package validation
 
 import (
+	"fmt"
 	"reflect"
+	"sync"
 
+	"google.golang.org/genproto/googleapis/api/metric"
 	"google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	StatusDuplicateMetricDescriptorName = status.New(codes.AlreadyExists, "metric descriptor with same name already exists")
```

Comment:
I think you can avoid exporting these, and in the tests just compare `codes.AlreadyExists` since we're moving away from comparing the error strings

Reply:
I was thinking the same thing, Done.